### DISCUSSION TITLE
Copy of a connection instance

### DISCUSSION THREAD

**Discussion by desertdognl** [bitfocus/companion#2887] [Ideas]
I often run live events with two computers running the same software. One main machine and one backup., Both machines run the same showfile of say Qlab. I now add both to Companion with their connection on their IP address.
This means that in programming, every button needs at least two actions. One command action to the first connection (main machine) and the same command to the second connection (backup machine)

It would be great if there would be an option to add a clone/copy/slave of a connection where all that is different is the IP address and all commands sent to the main connection are also sent to this clone/copy/slave. This way it's easy to add one or multiple backups/slaves without reprograming all buttons. Also when no backup is available it's easy to take it away and potentially have to reprogram all buttons to take the second command to the backup machine away which gives connection errors since it doesn't exist

**Comment by dnmeid** [comment]
This is an ages old feature request: #268
The solution I think of is a proxy module. That means you can set up some connections of the same type and then add as many as you want to the proxy connection. The proxy connection offers all actions of the individual connections, so you can either address an idividual connection or all via the proxy.
But it still is unclear how to handle feedbacks and this proxy connection would need to have access to the data exported by the individual connections, which means we have to change the interface and that is what prevents us from doing it. A lot of work, a lot of potential problems.

**Comment by TheAmazingReene** [comment]
Not sure if this has been considered, but I'm still looking for a nice way to swap between connections for commands. What would be really cool is if you could use a custom variable as the IP assignment and then could just have a Main/Backup button that changes the IP?

Or if the connection section of the Module could contain a Backup connection within it that you can enable or disable?
